app/src/main/java/org/kontalk/ui/view/ComposerBar.java

```diff
@@ -795,6 +801,35 @@ private void hideSoftKeyboard() {
             InputMethodManager.HIDE_IMPLICIT_ONLY);
     }
 
+    private Editable smileyConvert(Editable input){
+        if (containSmiley(input, ":)")) {
```

Comment:
Please use a hash map to store emoticon-emoji pairs. Also you potentially allocate (toString) and scan (contains+indexOf) the text twice. You could use the result of indexOf both to check for the string and to do the replace. Remember, we are on a restricted environment. It doesn't matter that modern smartphones have 16 cores and 4 GB of RAM: batteries are still fundamentally crap, so energy is precious.